core/services/ocr2/plugins/ocr2keeper/evm/abi.go

```diff
@@ -32,20 +31,30 @@ func NewEvmRegistryPackerV2_0(abi abi.ABI) *evmRegistryPackerV2_0 {
 	return &evmRegistryPackerV2_0{abi: abi}
 }
 
-func (rp *evmRegistryPackerV2_0) UnpackCheckResult(key types.UpkeepKey, raw string) (types.UpkeepResult, error) {
+func (rp *evmRegistryPackerV2_0) UnpackCheckResult(key ocr2keepers.UpkeepKey, raw string) (EVMAutomationUpkeepResult20, error) {
+	var (
+		result EVMAutomationUpkeepResult20
+	)
+
 	b, err := hexutil.Decode(raw)
 	if err != nil {
-		return types.UpkeepResult{}, err
+		return result, err
 	}
 
 	out, err := rp.abi.Methods["checkUpkeep"].Outputs.UnpackValues(b)
 	if err != nil {
-		return types.UpkeepResult{}, fmt.Errorf("%w: unpack checkUpkeep return: %s", err, raw)
+		return result, fmt.Errorf("%w: unpack checkUpkeep return: %s", err, raw)
 	}
 
-	result := types.UpkeepResult{
-		Key:   key,
-		State: types.Eligible,
+	block, id, err := splitKey(key)
+	if err != nil {
+		return result, err
+	}
+
+	result = EVMAutomationUpkeepResult20{
+		Block:    uint32(block.Uint64()),
+		ID:       id,
+		Eligible: true,
```

Comment:
nit. slightly confusing that we default to true and then later have an if which sets it to false. I know this behaviour existed before but it would be just simpler to not set it here and then later set `result.Eligible = upkeepNeeded` ?